src/Common/src/TypeSystem/Common/TargetDetails.cs

```diff
@@ -33,6 +33,22 @@ public enum TargetOS
     }
 
     /// <summary>
+    /// Specifies 
+    /// </summary>
+    public enum TargetApplicationModel
```

Comment:
Wasn't it called `TargetRuntime` in the previous iteration? I think it is much more appropriate name. E.g. if we started generating binaries for CoreCLR - it would be another member in this enum. CoreCLR certainly is not application model.